targets/csharp/source/source/PlayFabGameServerFileLogger.cs

```diff
@@ -0,0 +1,83 @@
+﻿// GSDK Only has the following supported frameworks for now
+#if NETSTANDARD2_0 || NETCOREAPP2_1
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace PlayFab
+{
+    public interface ILogger
+    {
+        void Start();
+        void Stop();
+        void Log(string message);
+    }
+
+    public static class LoggerFactory
+    {
+        public static ILogger CreateInstance(string logFolder)
+        {
+            return new FileSystemLogger(logFolder);
+        }
+    }
+
+    public class FileSystemLogger : ILogger
+    {
+        private StreamWriter _logWriter;
+        private string _logFolder;
+
+        public FileSystemLogger(string logFolder)
+        {
+            _logFolder = logFolder;
+        }
+
+        public void Log(string message)
+        {
+            if (_logWriter == null)
+            {
+                return; // Logging is disabled
+            }
+
+            _logWriter.WriteLine($"{DateTime.UtcNow.ToString("o")}\t{message}");
+            _logWriter.Flush();
+        }
+
+        public void Start()
+        {
+            if (_logWriter != null)
+            {
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(_logFolder))
+            {
+                _logFolder = Environment.CurrentDirectory;
+            }
+
+            try
+            {
+                if (!Directory.Exists(_logFolder))
+                {
+                    Directory.CreateDirectory(_logFolder);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logFolder = Environment.CurrentDirectory;
```

Comment:
Why do you change the logFolder and then throw the exception again?

Reply:
that's a good catch! now I'm not sure what the best thing to do here is... maybe I'll let it throw an exception

---
In reply to: [256212528](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256212528) [](ancestors = 256212528)